### ISSUE TITLE
Need to update plugin for 4.18 compatibility.

### ISSUE THREAD

**Issue by reapazor** [dotBunny/CLionSourceCodeAccess#69]
[6/8] Compile Module.CLionSourceCodeAccess.gen.cpp
In file included from /Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Intermediate/Build/Mac/UE4Editor/Development/CLionSourceCodeAccess/Module.CLionSourceCodeAccess.cpp:2:
/Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSettings.cpp:111:90: warning: 'GameDir' is deprecated: FPaths::GameDir() has been superseded by FPaths::ProjectDir(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile. [-Wdeprecated-declarations]
        this->CachedCMakeListPath = FPaths::Combine(*FPaths::ConvertRelativePathToFull(*FPaths::GameDir()),
                                                                                                ^
Runtime/Core/Public/Misc/Paths.h:117:29: note: 'GameDir' has been explicitly marked deprecated here
        static FORCEINLINE FString GameDir() { return ProjectDir(); }
                                   ^
In file included from /Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Intermediate/Build/Mac/UE4Editor/Development/CLionSourceCodeAccess/Module.CLionSourceCodeAccess.cpp:4:
/Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSourceCodeAccessor.cpp:65:68: warning: 'GameDir' is deprecated: FPaths::GameDir() has been superseded by FPaths::ProjectDir(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile. [-Wdeprecated-declarations]
        FString ProjectPath = *FPaths::ConvertRelativePathToFull(*FPaths::GameDir());
                                                                          ^
Runtime/Core/Public/Misc/Paths.h:117:29: note: 'GameDir' has been explicitly marked deprecated here
        static FORCEINLINE FString GameDir() { return ProjectDir(); }
                                   ^
In file included from /Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Intermediate/Build/Mac/UE4Editor/Development/CLionSourceCodeAccess/Module.CLionSourceCodeAccess.cpp:4:
/Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSourceCodeAccessor.cpp:551:83: warning: 'GameDir' is deprecated: FPaths::GameDir() has been superseded by FPaths::ProjectDir(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile. [-Wdeprecated-declarations]
                                             *FPaths::ConvertRelativePathToFull(*FPaths::GameDir()), LineNumber, *FullPath);
                                                                                         ^
Runtime/Core/Public/Misc/Paths.h:117:29: note: 'GameDir' has been explicitly marked deprecated here
        static FORCEINLINE FString GameDir() { return ProjectDir(); }
                                   ^
In file included from /Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Intermediate/Build/Mac/UE4Editor/Development/CLionSourceCodeAccess/Module.CLionSourceCodeAccess.cpp:4:
/Users/reapazor/Workspaces/dotBunny/Dethol/Game/Project/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSourceCodeAccessor.cpp:582:99: warning: 'GameDir' is deprecated: FPaths::GameDir() has been superseded by FPaths::ProjectDir(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile. [-Wdeprecated-declarations]
        const FString Path = FString::Printf(TEXT("\"%s\""), *FPaths::ConvertRelativePathToFull(*FPaths::GameDir()));
                                                                                                         ^
Runtime/Core/Public/Misc/Paths.h:117:29: note: 'GameDir' has been explicitly marked deprecated here
        static FORCEINLINE FString GameDir() { return ProjectDir(); }
                                   ^
4 warnings generated.

**Comment by tanis2000** [comment]
The 4.18 branch is still not working btw

```
Performing 2 actions (4 in parallel)
[1/2] Compile Module.CLionSourceCodeAccess.cpp
In file included from /Users/tanis/Documents/Unreal Projects/Test2DPlatformer/Plugins/CLionSourceCodeAccess/Intermediate/Build/Mac/UE4Editor/Development/CLionSourceCodeAccess/Module.CLionSourceCodeAccess.cpp:3:
In file included from /Users/tanis/Documents/Unreal Projects/Test2DPlatformer/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSourceCodeAccessModule.cpp:8:
/Users/tanis/Documents/Unreal Projects/Test2DPlatformer/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSourceCodeAccessModule.h:23:27: error: field type 'FCLionSourceCodeAccessor' is an abstract class
        FCLionSourceCodeAccessor CLionSourceCodeAccessor;
                                 ^
Developer/SourceCodeAccess/Public/ISourceCodeAccessor.h:58:15: note: unimplemented pure virtual method 'OpenSolutionAtPath' in 'FCLionSourceCodeAccessor'
        virtual bool OpenSolutionAtPath(const FString& InSolutionPath) = 0;
                     ^
Developer/SourceCodeAccess/Public/ISourceCodeAccessor.h:64:15: note: unimplemented pure virtual method 'DoesSolutionExist' in 'FCLionSourceCodeAccessor'
        virtual bool DoesSolutionExist() const = 0;
                     ^
In file included from /Users/tanis/Documents/Unreal Projects/Test2DPlatformer/Plugins/CLionSourceCodeAccess/Intermediate/Build/Mac/UE4Editor/Development/CLionSourceCodeAccess/Module.CLionSourceCodeAccess.cpp:3:
/Users/tanis/Documents/Unreal Projects/Test2DPlatformer/Plugins/CLionSourceCodeAccess/Source/CLionSourceCodeAccess/Private/CLionSourceCodeAccessModule.cpp:13:1: error: cannot initialize return object of type 'IModuleInterface *' with an rvalue of type 'FCLionSourceCodeAccessModule *'
IMPLEMENT_MODULE(FCLionSourceCodeAccessModule, CLionSourceCodeAccess);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Runtime/Core/Public/Modules/ModuleManager.h:733:11: note: expanded from macro 'IMPLEMENT_MODULE'
                        return new ModuleImplClass(); \
                               ^~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

**Comment by tanis2000** [comment]
@reapazor have a look at my PR here https://github.com/dotBunny/CLionSourceCodeAccess/pull/79
I don't know what that new method `DoesSolutionExist` should actually do. What's the solution file supposed to be in that context?

**Comment by reapazor** [comment]
I assume its if the project has generated files or not.

**Comment by tanis2000** [comment]
yes, it makes sense, but in this context what are we supposed to check? If we have already written the CMakeLists.txt file? What if that file is stale? Would we still return true? I'm not sure of the meaning of this method tbh

**Comment by reapazor** [comment]
I think it was 'WIP' where I was gonna check if it was stale/etc, but the collaboration took precedence.
